### PR TITLE
feat: align trpc usage to official docs

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

`t` is often used for translation stuff so importing it as your trpc object may feel weird. This is another way of using trpc v10 in that we define all our "base"-procedures and stuff in one file and import those, instead of the entire `t`-object.

Goes inline with the authedProcedure too btw.

Leave your thoughts, I'm just putting it out there. Happy to close if y'all don't like it.

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

**Comment by c-ehrlich** [comment]
I'd lean towards leaving it as is, mostly for the sake of keeping things consistent with the tRPC docs. The namespace collision of having just `t` can be fixed by the user by exporting `initTRPC.context` as any other name.

**Review by c-ehrlich — COMMENTED** [review]
I'd lean towards leaving it as is, mostly for the sake of keeping things consistent with the tRPC docs. The namespace collision of having just t can be fixed by the user by exporting initTRPC.context as any other name.

**Comment by juliusmarminge** [comment]
> I'd lean towards leaving it as is, mostly for the sake of keeping things consistent with the tRPC docs. The namespace collision of having just `t` can be fixed by the user by exporting `initTRPC.context` as any other name.

This is now the recommended way of doing things: https://trpc.io/docs/v10/nextjs#3-create-a-trpc-router

**Comment by nexxeln** [comment]
Can I also review 🥺

File: cli/template/addons/trpc/auth-server-utils.ts [hunk 1]
```diff
@@ -9,19 +9,30 @@ const t = initTRPC.context<Context>().create({
...
+/**
+ * Reusable middleware to ensure
+ * users are logged in
+ */
+const isAuthed = t.middleware(({ ctx, next }) => {
   if (!ctx.session || !ctx.session.user) {
     throw new TRPCError({ code: "UNAUTHORIZED" });
   }
   return next({
     ctx: {
```
Comment (c-ehrlich) [inline cli/template/addons/trpc/auth-server-utils.ts:28]: why is ctx not being spread in here anymore?

**Review by c-ehrlich — COMMENTED** [review]
I love importing the types directly instead of an object that has them.

Still getting used to getting rid of `t` in application code - I feel like using `t.procedure` etc in makes it very easy to see that the "base" version is being used. But we should go with tRPC docs over my opinions :)

File: cli/template/addons/trpc/auth-server-utils.ts (hunk 1 above)
Comment (juliusmarminge) [reply cli/template/addons/trpc/auth-server-utils.ts:28]: Its not needed in v10, its doing an override so only the stuff you want to change need to be explicit

File: cli/template/addons/trpc/auth-server-utils.ts [hunk 2]
```diff
@@ -9,19 +9,30 @@ const t = initTRPC.context<Context>().create({
...
 });
 
-export const createRouter = t.router;
+export const router = t.router;
 
-export const baseProcedure = t.procedure;
+/**
+ * Unprotected procedure
+ **/
+export const publicProcedure = t.procedure;
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply cli/template/addons/trpc/auth-server-utils.ts:17]: Where? It should be public everywhere

[the remark this answers is not in the document]
Comment (c-ehrlich) [reply cli/template/addons/trpc/auth-server-utils.ts:17]: Literally the one place I looked and nowhere else (nextjs/introduction.md) -_-

will open a PR on tRPC for that, but for ct3a it's all good then :)

**Review by c-ehrlich — APPROVED** [review]

**Pull request merged**